pkg/services/sqlstore/migrations/ualert/tables.go

```diff
@@ -12,34 +14,45 @@ const DefaultFieldMaxLength = 190
 // UIDMaxLength is the standard size for fields that contain UIDs.
 const UIDMaxLength = 40
 
-// AddMigration defines database migrations.
+// AddTablesMigrations defines database migrations that affect Grafana Alerting tables.
 func AddTablesMigrations(mg *migrator.Migrator) {
-	AddAlertDefinitionMigrations(mg, 60)
-	AddAlertDefinitionVersionMigrations(mg)
+	// Migrations are meant to be immutable, any modifications to table structure
+	// should come in the form of a new migration appended to the end of AddTablesMigrations
+	// instead of modifying an existing one. This ensure that tables are modified in a consistent and correct order.
+	historicalTableMigrations(mg)
+
+	extractAlertmanagerConfigurationHistoryMigration(mg)
```

Comment:
Thoughts on just combining this func with `historicalTablesMigrations` and having one big non-frozen func containing everything that we append to?

Alternatively, this one could go in `historicalTablesMigrations` - This one is kind of paired logically with `addAlertmanagerConfigHistoryMigrations(mg)` so IMO they should live next to each other, wherever they go.

Reply:
The intention was to use the latest one as a sort of example to follow, but I agree it should be alongside `addAlertmanagerConfigHistoryMigrations`. I'll move it into `historicalTablesMigrations`